tomcat-mixin/dashboards/tomcat-overview.libsonnet

```diff
@@ -0,0 +1,767 @@
+      'sum(tomcat_threadpool_currentthreadcount{job=~"$job", instance=~"$instance", protocol=~"$protocol", port=~"$port"}) by (job, instance)',
+      datasource=promDatasource,
+      legendFormat='{{job}} - {{instance}} - active total',
+    ),
+    prometheus.target(
+      'tomcat_threadpool_connectioncount{job=~"$job", instance=~"$instance", protocol=~"$protocol", port=~"$port"}',
+      datasource=promDatasource,
+      legendFormat='{{job}} - {{instance}} - {{protocol}}-{{port}} - connections',
+    ),
+    prometheus.target(
+      'tomcat_threadpool_pollerthreadcount{job=~"$job", instance=~"$instance", protocol=~"$protocol", port=~"$port"}',
+      datasource=promDatasource,
+      legendFormat='{{job}} - {{instance}} - {{protocol}}-{{port}} - poller',
+    ),
+    prometheus.target(
+      'tomcat_threadpool_keepalivecount{job=~"$job", instance=~"$instance", protocol=~"$protocol", port=~"$port"}',
+      datasource=promDatasource,
+      legendFormat='{{job}} - {{instance}} - {{protocol}}-{{port}} - idle',
+    ),
+    prometheus.target(
+      'tomcat_threadpool_currentthreadcount{job=~"$job", instance=~"$instance", protocol=~"$protocol", port=~"$port"}',
+      datasource=promDatasource,
+      legendFormat='{{job}} - {{instance}} - {{protocol}}-{{port}} - active',
+    ),
+  ],
+  type: 'timeseries',
+  title: 'Threads',
+  description: 'The number of various threads being used by a Tomcat connector',
+  fieldConfig: {
+    defaults: {
+      color: {
+        mode: 'palette-classic',
+      },
+      custom: {
+        axisCenteredZero: false,
+        axisColorMode: 'text',
+        axisLabel: '',
+        axisPlacement: 'auto',
+        barAlignment: 0,
+        drawStyle: 'line',
+        fillOpacity: 0,
+        gradientMode: 'none',
+        hideFrom: {
+          legend: false,
+          tooltip: false,
+          viz: false,
+        },
+        lineInterpolation: 'linear',
+        lineWidth: 1,
+        pointSize: 5,
+        scaleDistribution: {
+          type: 'linear',
+        },
+        showPoints: 'auto',
+        spanNulls: false,
+        stacking: {
+          group: 'A',
+          mode: 'none',
+        },
+        thresholdsStyle: {
+          mode: 'off',
+        },
+      },
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+          },
+          {
+            color: 'red',
+            value: 80,
+          },
+        ],
+      },
+    },
+    overrides: [],
+  },
+  options: {
+    legend: {
+      calcs: [],
+      displayMode: 'list',
+      placement: 'right',
+      showLegend: true,
+    },
+    tooltip: {
+      mode: 'single',
+      sort: 'none',
+    },
+  },
+};
+
+local logsPanel = {
+  datasource: lokiDatasource,
+  targets: [
+    {
+      datasource: lokiDatasource,
+      editorMode: 'code',
+      expr: '{filename="/var/log/tomcat9/catalina.out", job=~"$job", instance=~"$instance"} |= ``',
```

Comment:
@schmikei do we do anything special when there are multiple possible locations? do we ever use a variable or just pick one like this?

Reply:
I think we can use wildcard matches if there is more to match i.e.

```
filename="~/var/log/tomcat.*/catalina.out"
```

if that is what you mean? Otherwise if it is always that path we can use the strict match.

Reply:
@schmikei I think there were some drastically different options for the log location

Reply:
Seems like ```__path__: file1, file2, file3``` is working. Will ask Grafana about preference on displaying multiple log outputs in one panel.